plugins/basic/python/tk_maya_basic/plugin_engine.py

```diff
@@ -44,10 +44,31 @@ def bootstrap(sg_user, progress_callback, completed_callback, failed_callback):
     # Create a boostrap manager for the logged in user with the plug-in configuration data.
     toolkit_mgr = sgtk.bootstrap.ToolkitManager(sg_user)
 
-    # pass the manager to the manifest for basic init
+    # Pass the boostrap manager to the manifest for basic initialization.
     manifest.initialize_manager(toolkit_mgr, PLUGIN_ROOT_PATH)
 
-    # check if the target core supports async init
+    # Retrieve the Shotgun entity type and id when they exist in the environment.
+    entity_type = os.environ.get("SHOTGUN_ENTITY_TYPE")
+    entity_id = os.environ.get("SHOTGUN_ENTITY_ID")
+
+    # The entity id must be an integer number.
+    try:
+        entity_id = int(entity_id)
+    except ValueError:
+        logger.error("Environment variable SHOTGUN_ENTITY_ID value '%s' is not an integer number. "
+                     "Shotgun will be initialized in site context." % entity_id)
+        entity_id = None
+
+    if entity_type and entity_id:
+        # Set the entity to launch the engine for.
+        entity = {"type": entity_type, "id": entity_id}
+    else:
+        # Set the entity to launch the engine in site context.
+        entity = None
```

Comment:
Done, when `(entity_type and not entity_id) or (not entity_type and entity_id)`.